packages/synapse-react-client/src/components/EntityTreeTable/hooks/useEntityTreeState.ts

```diff
@@ -73,36 +83,72 @@ export const useEntityTreeState = (
     }
   }, [sorting])
 
+  const rootChildrenRequest: EntityChildrenRequest = useMemo(() => {
+    const baseRequest: EntityChildrenRequest = {
+      parentId: rootId,
+      includeTypes,
+    }
+
+    if (sortBy !== undefined && sortDirection !== undefined) {
+      return {
+        ...baseRequest,
+        sortBy,
+        sortDirection,
+      }
+    }
+
+    return baseRequest
+  }, [rootId, sortBy, sortDirection])
+
+  const rootChildrenQueryKey = useMemo(
+    () => keyFactory.getEntityChildrenQueryKey(rootChildrenRequest, false),
+    [keyFactory, rootChildrenRequest],
+  )
+
   // Reset tree data when sorting changes
   const resetTreeData = useCallback(() => {
+    void queryClient.removeQueries({
+      queryKey: rootChildrenQueryKey,
+      exact: true,
+    })
```

Comment:
Why do we need to `removeQueries`? Never used that API

Reply:
It's my attempt at removing cached results on `resetTreeData()` without triggering the new subscription to `invalidate` events
If the `useGetEntityChildrenInfinite` approach works, then I wouldn't need any of this complexity though